internal/pkg/dev/minispire/grpc-authinfo.go

```diff
@@ -0,0 +1,116 @@
+package minispire
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	ErrInvalidConnection = errors.New("invalid connection")
+)
+
+type Conn struct {
+	net.Conn
+	Info AuthInfo
+}
+
+func (c *Conn) Close() error {
+	return c.Conn.Close()
+}
+
+type grpcCredentials struct{}
+
+func NewCredentials() credentials.TransportCredentials {
+	return &grpcCredentials{}
+}
+
+func (c *grpcCredentials) ClientHandshake(_ context.Context, _ string, conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	conn.Close()
+	return conn, AuthInfo{}, ErrInvalidConnection
+}
+
+func (c *grpcCredentials) ServerHandshake(conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	wrappedCon, ok := conn.(*net.UnixConn)
+	if !ok {
+		conn.Close()
+		log.Printf("invalid connection type: %T", conn)
+		return conn, AuthInfo{}, ErrInvalidConnection
+	}
+	// get the caller's PID, UID, and GID
+	sys, err := wrappedCon.SyscallConn()
+	if err != nil {
+		log.Printf("unable to get peer credentials: %v", err)
+		conn.Close()
+		return conn, AuthInfo{}, ErrInvalidConnection
+	}
+
+	auth := AuthInfo{}
+	sys.Control(func(fd uintptr) {
+		cred, err := syscall.GetsockoptUcred(int(fd), syscall.SOL_SOCKET, syscall.SO_PEERCRED)
```

Comment:
FYI, just adding a comment here related to this [issue](https://github.com/cofide/cofide-cli/issues/171) tracked in `cofide-cli`.